src/script/storage/record/EventRecord.ts

```diff
@@ -40,13 +42,22 @@ export interface AssetRecord {
 export type UserReactionMap = {[userId: string]: ReactionType};
 
 /** represents an event that was saved to the DB */
-type StoredEvent = {
+export type StoredEvent<T> = {
   primary_key: string;
   category: number;
   id: string;
+  /** if the message is ephemeral, that's the amount of time it should be displayed to the user */
+  ephemeral_expires?: number | boolean | string;
```

Comment:
why ephemeral_expires can be 3 different types?

Reply:
This is legacy sadly. 
It can mean:
- it's a datestring (string)
- it's a timestamp (number)
- it has been consumed (boolean)

Reply:
Actually let me add this as a comment :) Good call